src/transactions/apiTestHelpers.ts

```diff
@@ -0,0 +1,45 @@
+import type { EnhancedStore, Middleware, Reducer, UnknownAction } from '@reduxjs/toolkit'
+import { combineReducers, configureStore } from '@reduxjs/toolkit'
+import { ApiReducersKeys } from 'src/redux/apiReducersList'
+import { RootState } from 'src/redux/reducers'
+import { RecursivePartial } from 'test/utils'
+
+// https://medium.com/@johnmcdowell0801/testing-rtk-query-with-jest-cdfa5aaf3dc1
+export function setupApiStore<
```

Comment:
Could we use/extend our existing mockStore instead?

https://github.com/valora-inc/wallet/blob/1aa94d6e31ed8dcd9baffa26b3682dc800b2bc80/test/utils.ts#L76-L82

So it's not specific to this reducer.

This way we'll respect one of the Redux testing principle: 
> Prefer writing integration tests with everything working together.

https://redux.js.org/usage/writing-tests

Reply:
@jeanregisser Initially, I've tried to use this store and I couldn't get it to work. 

To test RTK-Query it is required to have api middlewares in the mocked store. `configureMockStore` accepts an array of middlewares as an argument and I've tried to pass it was triggering for all the other tests to fail with error saying something similar to: "Actions must not be dispatched but passed as a plain object". Considering, that RTK-Query was released in 2021 and the last update for `redux-mock-store` was in 2018, it seems like there were no changes made to the library to make is able to work with RTK-Query. 

I was looking for the solution and found [this article](https://medium.com/@johnmcdowell0801/testing-rtk-query-with-jest-cdfa5aaf3dc1) (I've also put it in a comment there so there's some reference) that pointed me in a direction to try to use the [helper function](https://github.com/reduxjs/redux-toolkit/blob/e7540a5594b0d880037f2ff41a83a32c629d3117/packages/toolkit/src/tests/utils/helpers.tsx#L186) Redux team uses themselves to create a testable store. I've slightly modified it but it seemed to do the job. 

P.S. Here's the quote from the article explaining the potential reason of why `redux-mock-store` doesn't work with RTK-Query:
_"Typically we might reach for a tool like [redux-mock-store](https://github.com/reduxjs/redux-mock-store) but this library doesn’t work when testing RTK Query because dispatching RTK Query actions relies on there being a reducer for the API service and redux-mock-store doesn’t allow the use of reducers."_

Reply:
@jeanregisser I've now added all the data from [getMockStoreData](https://github.com/valora-inc/wallet/blob/main/test/utils.ts#L84) to it so it's identical. Sorry, definitely missed this initially!
The two mocked stores should be identical now in terms of preloaded data.